test/Authentication/DefaultAuthenticationListenerTest.php

```diff
@@ -803,4 +803,49 @@ public function testOauth2RequestIncludesHeaders()
         $this->listener->attach(new OAuth2Adapter($server));
         $this->listener->__invoke($this->mvcAuthEvent);
     }
+
+    /**
+     * @group 83
+     */
+    public function testAllowsAdaptersToReturnResponsesAndReturnsThemDirectly()
```

Comment:
This test ensures that the authentication listener will directly return a response instance returned by an adapter.